lib/config/folderconfiguration.go

```diff
@@ -124,12 +124,20 @@ func (f *FolderConfiguration) CreateMarker() error {
 // CheckPath returns nil if the folder root exists and contains the marker file
 func (f *FolderConfiguration) CheckPath() error {
 	fi, err := f.Filesystem().Stat(".")
-	if err != nil || !fi.IsDir() {
+	if err != nil {
+		l.Debugln("folder root:", err) // Log the error for debugging purposes
+		return errPathMissing
+	}
+
+	// Users might have the root directory as a symlink or reparse point.
+	// Furthermore, OneDrive bullcrap uses a magic reparse point to the cloudz...
+	if !fi.IsDir() && !fi.IsSymlink() {
 		return errPathMissing
```

Comment:
Given that there is now a separate check I think it would be nice to have an `errPathNotDir` or something that is more descriptive than "missing", as users will not consider it missing.

Maybe even check if the first error `os.IsNotExist` and only then return `errPathMissing`, otherwise the error as is? In the name of getting a possibly more concrete and specific error in front of the user.

Oh, and how about a test for this? Just something that creates a directory, creates a symlink to it, and verifies that checkpath on the symlink succeeds. If goos=="windows" and the symlink create fails, t.Skip